File: SafeVote/update_passwords.js
const { MongoClient } = require('mongodb');

const uri = "mongodb+srv://stharunkumar069_db_user:Tharun%4018@cluster0.frcnaxx.mongodb.net/safevote?retryWrites=true&w=majority&appName=Cluster0";

async function updatePasswords() {
  const client = new MongoClient(uri);
  try {
    await client.connect();
    console.log("Connected to MongoDB Atlas!");

    const db = client.db("safevote");
    const result = await db.collection("students").updateMany(
      {},
      { $set: { password: "SIET" } }
    );

    console.log(`Matched: ${result.matchedCount} students`);
    console.log(`Updated: ${result.modifiedCount} students`);
    console.log("Done! All passwords changed from 'atkboss' to 'SIET'");
  } catch (err) {
    console.error("Error:", err.message);
  } finally {
    await client.close();
  }
}

updatePasswords();

File: SafeVote/backend/elections/management/commands/update_passwords.py
from django.core.management.base import BaseCommand
from utils.database import get_students_collection
from utils.security import hash_password

class Command(BaseCommand):
    help = "Update student passwords matching update_passwords.js"

    def add_arguments(self, parser):
        parser.add_argument('--password', type=str, default='SIET', help='Password to set for all students')
        parser.add_argument('--hash', action='store_true', help='Store password as secure PBKDF2 hash instead of plaintext')

    def handle(self, *args, **options):
        new_pass = options['password']
        should_hash = options['hash']

        stored_value = hash_password(new_pass) if should_hash else new_pass

        self.stdout.write("Connected to MongoDB for password update...")
        students_col = get_students_collection()
        result = students_col.update_many(
            {},
            {"$set": {"password": stored_value}}
        )

        self.stdout.write(f"Matched: {result.matched_count} students")
        self.stdout.write(f"Updated: {result.modified_count} students")
        self.stdout.write(self.style.SUCCESS(f"Done! Student passwords updated."))
